src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/CommandFakePlayer.kt

```diff
@@ -0,0 +1,314 @@
+
+    /**
+     * Stores all fake players.
+     */
+    private var fakePlayers: MutableList<FakePlayer>? = null
+
+    private var recording = false
+    private var snapshots: MutableList<PosPoseSnapshot>? = null
+
+    fun createCommand(): Command {
+        return CommandBuilder
+            .begin("fakeplayer")
+            .hub()
+            .subcommand(spawnCommand())
+            .subcommand(removeCommand())
+            .subcommand(clearCommand())
+            .subcommand(startRecordingCommand())
+            .subcommand(endRecordingCommand())
+            .build()
+    }
+
+    private fun spawnCommand() = CommandBuilder
+        .begin("spawn")
+        .parameter(
+            ParameterBuilder
+                .begin<String>("name")
+                .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                .optional()
+                .build()
+        )
+        .handler { _, args ->
+            spawn(args, false)
+        }
+        .build()
+
+    private fun removeCommand() = CommandBuilder
+        .begin("remove")
+        .parameter(
+            ParameterBuilder
+                .begin<String>("name")
+                .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                .optional()
+                .build()
+        )
+        .handler { command, args ->
+            checkInGame()
+
+            if (fakePlayers == null || fakePlayers!!.isEmpty()) {
+                throw CommandException(translation("liquidbounce.command.fakeplayer.noFakePlayers"))
+            }
+
+            val name = args.getOrNull(0)?.toString() ?: "FakePlayer"
+            var anyRemoved = false
+
+            fakePlayers?.removeIf { fakePlayer ->
+                @Suppress("ReplaceCallWithBinaryOperator")
+                val remove = name.equals(fakePlayer.name.string)
+                if (remove) {
+                    world.removeEntity(fakePlayer.id, Entity.RemovalReason.KILLED)
+                    chat(
+                        regular(
+                            command.result(
+                                "fakePlayerRemoved",
+                                BigDecimal(fakePlayer.x).setScale(1, RoundingMode.HALF_UP).toDouble(),
+                                BigDecimal(fakePlayer.y).setScale(1, RoundingMode.HALF_UP).toDouble(),
+                                BigDecimal(fakePlayer.z).setScale(1, RoundingMode.HALF_UP).toDouble()
+                            )
+                        )
+                    )
+                    anyRemoved = true
+                    return@removeIf true
+                } else {
+                    return@removeIf false
+                }
+            }
+
+            if (!anyRemoved) {
+                chat(warning((command.result("noFakePlayerNamed", name))))
+                chat(regular(command.result("currentlySpawned")))
+                fakePlayers!!.forEach { fakePlayer ->
+                    chat(regular("- " + fakePlayer.name.string))
+                }
+            }
```

Comment:
This code has several problems

## Design Problem
`removeIf` expects a predicate which should have _no side-effects_. This could cause problems like sending messages twice or crashing due to removing the same entity twice. This code is called a few times in the programs run-time, performance does not really matter so it is not worth trading it for code safety. 

Better solution:
```Kotlin
val playersToRemove = fakePlayers.filter { fakePlayer.name.string == name }

if (playersToRemove.isEmpty()) {
    // Warn player
    return@handler
}

playersToRemove.forEach { player ->
    // Remove from world
    // Notify user
}
```
## Smaller Problems

> ```Kotlin
> @Suppress("ReplaceCallWithBinaryOperator")
> val remove = name.equals(fakePlayer.name.string)
> ```
IntelliJ told you that this code smells for a reason.

You can replace this with `name == fakePlayer.name.string` (which is equivalent)

Try to fix compiler warnings instead of suppressing them

-----
> ```Kotlin
> BigDecimal(fakePlayer.z).setScale(1, RoundingMode.HALF_UP).toDouble()
> ```

What are you trying to do? Rounding the number? Mayne `round(...)` is a better option.

If you know why you used this please add a comment with that reason.

-----
> ```Kotlin
> var anyRemoved = false
> ...
> anyRemoved = true
> ...
> if (!anyRemoved) {...}
> ```
`removeIf` already returns `true` if any elements were deleted (see [doc](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#removeIf-java.util.function.Predicate-))

Reply:
ah alr i'll change that

Reply:
I didn't use the ``MathJVM#round`` function to have one decimal place.

Reply:
> This code has several problems
> 
> ## Design Problem
> `removeIf` expects a predicate which should have _no side-effects_. This could cause problems like sending messages twice or crashing due to removing the same entity twice. This code is called a few times in the programs run-time, performance does not really matter so it is not worth trading it for code safety.
> 
> Better solution:
> 
> ```kotlin
> val playersToRemove = fakePlayers.filter { fakePlayer.name.string == name }
> 
> if (playersToRemove.isEmpty()) {
>     // Warn player
>     return@handler
> }
> 
> playersToRemove.forEach { player ->
>     // Remove from world
>     // Notify user
> }
> ```
> 
> ## Smaller Problems
> > ```kotlin
> > @Suppress("ReplaceCallWithBinaryOperator")
> > val remove = name.equals(fakePlayer.name.string)
> > ```
> 
> IntelliJ told you that this code smells for a reason.
> 
> You can replace this with `name == fakePlayer.name.string` (which is equivalent)
> 
> Try to fix compiler warnings instead of suppressing them
> 
> > ```kotlin
> > BigDecimal(fakePlayer.z).setScale(1, RoundingMode.HALF_UP).toDouble()
> > ```
> 
> What are you trying to do? Rounding the number? Mayne `round(...)` is a better option.
> 
> If you know why you used this please add a comment with that reason.
> 
> > ```kotlin
> > var anyRemoved = false
> > ...
> > anyRemoved = true
> > ...
> > if (!anyRemoved) {...}
> > ```
> 
> `removeIf` already returns `true` if any elements were deleted (see [doc](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#removeIf-java.util.function.Predicate-))

this is pure insanity